### ISSUE TITLE
TypeError: only pages can be inserted with version 2.6.0

### ISSUE THREAD

**Issue by jeromerobert** [pikepdf/pikepdf#174]
With pikepdf 2.6.0 the following program:

```python
nd = pikepdf.Pdf.new()
np = pikepdf.Dictionary(Type=pikepdf.Name.Page)
nd.pages.append(np)
```

raise this error:

```
    nd.pages.append(np)
TypeError: only pages can be inserted
```

It was working with version 2.5.2 and before.

Context: I'm using this pattern [here](https://github.com/pdfarranger/pdfarranger/blob/1.7.0/pdfarranger/exporter.py#L93) to support page scaling. Parent bug: https://github.com/pdfarranger/pdfarranger/issues/427

**Comment by jbarlow83** [comment]
This issue was triggered by a change in libqpdf 10.2.0, and will occur when any version of pikepdf is used with libqpdf 10.2.0 or later. That does mean users of the pikepdf wheels are affected because the wheel bundles libqpdf. But it's unrelated to the specific pikepdf version.

pikepdf 2.7.0 will adjust for changes in how libqpdf verifies that page objects.

**Comment by dreua** [comment]
Hey, as always, huge thanks for the blazing fast fix and pikepdf in general! :heart: 

Unfortunately we have [some fall out](https://github.com/pdfarranger/pdfarranger/issues/427) from this issue and I'm looking into ways to properly fix it for our users. I really don't want to point fingers here or throw more work at you or the qpdf developer (huge thanks to you too!), but I think for a proper fix we first need to know where exactly the issue was introduced. I came up with options:
1. pdfarranger is using pikepdf in an unsupported or really unusal way (:arrow_right: we need to fix it)
2. pikepdf is using qpdf in an unsupported way (:arrow_right: Would a backport of the 2.7.0 fix on 1.19 be reasonable? Or should we rather convince distributions to upgrade pikepdf major version in their stable releases?)
3. qpdf broke its api in version 10.2.0

In Fedora I could stop the qpdf update in testing but the [pikepdf package maintainer can not simply update 1.x to 2.x in a stable release due to distribution rules](https://bugzilla.redhat.com/show_bug.cgi?id=1933531). Manjaro/Arch user repos [seem to have a similar issue](https://github.com/pdfarranger/pdfarranger/issues/427#issuecomment-791875309).

**Comment by jbarlow83** [comment]
The issue is an interaction between pikepdf and qpdf. The change in qpdf makes sense and I would not consider the API broken. qpdf uses pikepdf in its release process to check for regressions. pikepdf would have needed full branch coverage of the qpdf APIs it uses to find this issue.

Page objects must be indirect objects. qpdf's `isPageObject()` function now tests for that. pikepdf normally promotes a direct object offered as a page to an indirect object, because it eliminates boilerplate. (For other reasons, I should overhaul the page API so that pikepdf presents pages as `pikepdf.Page` rather than `pikepdf.Dictionary`; but that will be for another time.)

You could do the following in pdfarranger wherever a page is created:

```python

nd = pikepdf.Pdf.new()
np = pikepdf.Dictionary(Type=pikepdf.Name.Page)
if pikepdf.__version__ < '2.7.0':
    np = nd.make_indirect(np)
 nd.pages.append(np)

```

(You could do `.make_indirect()` unconditionally rather than making it a shim, but you don't have to. pikepdf will make it indirect if needed.)

I did try cherrypicking the fix back to v1.19.0, but I found I actually had to cherry pick half a dozen other things to get all the tests green, and it would need more testing besides, e.g. to ensure compatibility with older qpdf. I think it will be less work and risk for all concerned if we try shimming this in pdfarranger first. If similar problems crop up elsewhere, I'll do a new pikepdf v1.x release.

A corollary of those half dozen other changes is that there are other places where the combination of pikepdf v1.x and qpdf 10.2.0+ may not behave as documented, but they at least seem more minor.

I think the easiest place to make a change is pdfarranger at the moment.

**Comment by dreua** [comment]
Great I'll try to implement this fix tonight!
